### PR TITLE
"Fly Free, My Pretty" solo duty

### PR DESCRIPTION
**Pull request by xanunderscore**
added "first class" mount/RP support to rotation modules, i.e. RMM won't try to activate a module if you're mounted and it doesn't support queuing mounted actions (which currently none do). this will allow users to keep their existing presets activated when using AI mode in MSQ solo duties, since those presets will no longer interfere with whatever module-specific actions are queued

### CODE REVIEW

File: BossMod/Components/RoleplayModule.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    public abstract void Execute(Actor? primaryTarget);
+
+    public override void AddAIHints(int slot, Actor actor, PartyRolesConfig.Assignment assignment, AIHints hints)
+    {
+        _hints = hints;
+        _player = actor;
+
+        Execute(WorldState.Actors.Find(actor.TargetID) ?? Module.PrimaryActor);
+    }
```
Comment (awgil) [inline BossMod/Components/RoleplayModule.cs:19]: Why is this needed - why not just let components override AddAIHints directly? I don't really like saving these things as a persistent state in components, since it would allow accidentally using them at the wrong time

File: BossMod/Components/RoleplayModule.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    protected void UseGCD(Roleplay.AID action, Actor? target, Vector3 targetPos = default)
+        => Hints.ActionsToExecute.Push(ActionID.MakeSpell(action), target, ActionQueue.Priority.High, targetPos: targetPos);
+
+    protected void UseOGCD(Roleplay.AID action, Actor? target, Vector3 targetPos = default)
+        => Hints.ActionsToExecute.Push(ActionID.MakeSpell(action), target, ActionQueue.Priority.Low, targetPos: targetPos);
```
Comment (awgil) [inline BossMod/Components/RoleplayModule.cs:25]: A lot of this makes sense as utilities for rotation modules, and I appreciate that 'roleplay components' are kind of similar, but does it really make too much sense? Not like anyone is going to create a cdplan for rp battles, so eg priorities don't really need to adhere to the actionqueue conventions; don't think gcd/ogcd distinction is even too meaningful during rp battles...

File: BossMod/Autorotation/RotationModule.cs [hunk 3]
```diff
@@ -23,10 +23,12 @@ public enum RotationModuleQuality
 // the configuration part of the rotation module
 // importantly, it defines constraints (supported classes and level ranges) and strategy configs (with their sets of possible options) used by the module to make its decisions
 // rotation modules can optionally be constrained to a specific boss module, if they are used to implement custom encounter-specific logic - these would only be available in plans for that module
-public sealed record class RotationModuleDefinition(string DisplayName, string Description, string Author, RotationModuleQuality Quality, BitMask Classes, int MaxLevel, int MinLevel = 1, Type? RelatedBossModule = null)
+public sealed record class RotationModuleDefinition(string DisplayName, string Description, string Author, RotationModuleQuality Quality, BitMask Classes, int MaxLevel, int MinLevel = 1, Type? RelatedBossModule = null, bool CanUseWhileMounted = false, bool CanUseWhileRoleplaying = false)
```
[the remark this answers is not in the document]
Comment (xanunderscore) [reply BossMod/Autorotation/RotationModule.cs:26]: actually if all the RP stuff is going to be boss module specific, then no, these flags are pointless

File: BossMod/Components/RoleplayModule.cs (hunk 2 above)
Comment (xanunderscore) [reply BossMod/Components/RoleplayModule.cs:25]: the hien battle against zenos at the end of stb actually does have gcd vs ogcd distinction., not that it really matters, since that whole battle is a joke - the point is to stall against wol - but it is at least nice to have a function defined for it. i can remove these and just replace them with one function called `UseAction` for example if that would be better

File: BossMod/Components/RoleplayModule.cs (hunk 1 above)
Comment (xanunderscore) [reply BossMod/Components/RoleplayModule.cs:19]: these are just convenience functions so components can pass stuff to AIHints. `hints.ActionsToExecute.Push(ActionID.MakeSpell(Roleplay.AID.MyCoolAction), target, ActionQueue.Priority.VeryHigh)` is a little more verbose than `UseAction(Roleplay.AID.MyCoolAction, target)`

File: BossMod/Autorotation/RotationModule.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (xanunderscore) [reply BossMod/Autorotation/RotationModule.cs:26]: wait no, isn't there a vehicle mechanic in Rival Wings? also, Praetorium has the magitek armor section, and i'm not sure how a module would work for that, since there's not really a Boss and modules require one to be defined

[the remark this answers is not in the document]
Comment (awgil) [reply BossMod/Autorotation/RotationModule.cs:26]: I kinda don't see people including specific rotation module in their presets just for prae?

And module could still be defined for a trash I guess... Well, I think i'll kill it for now, and if we end up needing it later, we'll remake it.

[the remark this answers is not in the document]
Comment (xanunderscore) [reply BossMod/Autorotation/RotationModule.cs:26]: i was thinking more of one preset for "all vehicle combat" that would get included in the default presets list. would be useful not only for prae but also for the "magiteknical failure" aether current sidequest in Stormblood. and also wherever else vehicle combat is used, which is basically nowhere, but it's so simple it doesn't seem like that big of a deal to implement
